nixos/modules/host.nix

```diff
@@ -148,48 +70,38 @@ in
       {
         assertion =
           !(cfg.tls.enable && (cfg.tls.caCertPath == "" || cfg.tls.certPath == "" || cfg.tls.keyPath == ""));
-        message = "The TLS option requires paths' to CA certificate, service certificate, and service key.";
+        message = ''
+          The TLS configuration requires paths' to CA certificate, service certificate, and service key.
+          To disable TLS, set 'tls.enable = false;'.
+        '';
       }
     ];
 
-    systemd.services."givc-${cfg.name}" = {
+    systemd.services."givc-${cfg.agent.name}" = {
       description = "GIVC remote service manager for the host.";
       enable = true;
       after = [ "network-online.target" ];
       wants = [ "network-online.target" ];
       wantedBy = [ "multi-user.target" ];
-      # @TODO add requirement for admin to run before host module
       serviceConfig = {
         Type = "exec";
         ExecStart = "${givc-agent}/bin/givc-agent";
         Restart = "always";
         RestartSec = 1;
       };
-      environment =
-        {
-          "NAME" = "${cfg.name}";
-          "ADDR" = "${cfg.addr}";
-          "PORT" = "${cfg.port}";
-          "PROTO" = "${cfg.protocol}";
-          "DEBUG" = "${trivial.boolToString cfg.debug}";
-          "TYPE" = "0";
-          "SUBTYPE" = "1";
-          "TLS" = "${trivial.boolToString cfg.tls.enable}";
-          "SERVICES" = "${concatStringsSep " " cfg.services}";
-          "ADMIN_SERVER_NAME" = "${cfg.admin.name}";
-          "ADMIN_SERVER_ADDR" = "${cfg.admin.addr}";
-          "ADMIN_SERVER_PORT" = "${cfg.admin.port}";
-          "ADMIN_SERVER_PROTO" = "${cfg.admin.protocol}";
-        }
-        // attrsets.optionalAttrs cfg.tls.enable {
-          "CA_CERT" = "${cfg.tls.caCertPath}";
-          "HOST_CERT" = "${cfg.tls.certPath}";
-          "HOST_KEY" = "${cfg.tls.keyPath}";
-        };
+      environment = {
+        "AGENT" = "${toJSON cfg.agent}";
```

Comment:
I like the reduced number of environment variables; something about putting JSON in environment variables feels hackish; should we consider a full-on config file?

Reply:
yes, we should definitely move in that direction. Perhaps a dynamic config could be possible as well, e.g., as part of a registry response...